src/backend/distributed/transaction/transaction_management.c

```diff
@@ -408,11 +419,11 @@ CoordinatedSubTransactionCallback(SubXactEvent event, SubTransactionId subId,
 
 		case SUBXACT_EVENT_COMMIT_SUB:
 		{
-			PopSubXact(subId);
 			if (InCoordinatedTransaction())
 			{
 				CoordinatedRemoteTransactionsSavepointRelease(subId);
 			}
+			PopSubXact(subId);
```

Comment:
What about the code below?  (case SUBXACT_EVENT_ABORT_SUB)

Reply:
`ROLLBACK TO SAVEPOINT` never errors out, so in  `case SUBXACT_EVENT_ABORT_SUB` order doesn't matter, but changed it after our discussion to keep the order of pops consistent.